R/data.table.R

```diff
@@ -1029,8 +1029,14 @@ replace_dot_alias = function(e) {
           if (is.name(lhs)) {
             lhs = as.character(lhs)
           } else {
-            # e.g. (MyVar):= or get("MyVar"):=
-            lhs = eval(lhs, parent.frame(), parent.frame())
+            #i.e lhs is names(.SD) || setdiff(names(.SD), cols) || (cols)
+            replace_names_sd = function(e, cols){
```

Comment:
not essential to this PR, but we have a couple of these little helper functions to iterate over the AST and edit it somehow, perhaps we can write something like `ast_apply` that can be used more generally?

Reply:
I am not sure what AST is. I know @jangorecki has taken ```replace_dot_alias``` to C in his #4174 PR but it is not generalized.

Reply:
AST=abstract syntax tree (I actually had to look that up myself)

https://adv-r.hadley.nz/expressions.html

not something that needs to be handled in this PR, just something I wanted to write down for safe keeping